rspec-mode.el

```diff
@@ -513,21 +549,49 @@ in long-running test suites."
 (declare-function dired-get-marked-files "dired")
 
 (defun rspec-dired-verify ()
-  "Run all specs in the current directory."
+  "Run all specs in the current directory, filtered by one or multiple tags."
   (interactive)
   (rspec-run-single-file (dired-current-directory) (rspec-core-options)))
 
+(defun rspec-dired-verify-tags ()
+  "Run all specs in the current directory, filtered by one or multiple tags."
+  (interactive)
+  (let* ((tags-string (completing-read-multiple "Select tags (separated by comma): " rspec-tags-history))
+         (tags-list (if (stringp tags-string) (split-string tags-string ",\\s-*") tags-string)))
+    (rspec-add-tags-to-history tags-list)
+    (rspec-run-single-file (dired-current-directory)
+                           (concat (mapconcat #'(lambda (tag) (format " --tag %s" tag)) tags-list " ") " "
+                                   (rspec-core-options)))))
+
 (defun rspec-dired-verify-single ()
   "Run marked specs or spec at point (works with directories too)."
   (interactive)
   (rspec-compile (dired-get-marked-files)
                  (rspec-core-options)))
 
+(defun rspec-dired-verify-tags-single ()
+  "Run marked specs or spec at point, filtered by one or multiple tags (works with directories too)."
+  (interactive)
+  (let* ((tags-string (completing-read-multiple "Select tags (separated by comma): " rspec-tags-history))
+         (tags-list (if (stringp tags-string) (split-string tags-string ",\\s-*") tags-string)))
+    (rspec-add-tags-to-history tags-list)
+    (rspec-compile (dired-get-marked-files)
```

Comment:
Here's a way to make this shorter: change the value of `rspec-command-options` locally (in the `let` form) and then call the "base" function. Like this:

```el
(defun rspec-dired-verify-tags-single ()
  "Run marked specs or spec at point, filtered by one or multiple tags (works with directories too)."
  (interactive)
  (let* ((tags (completing-read-multiple "Select tags (separated by comma): " rspec-tags-history))
         (rspec-command-options (concat rspec-command-options
                                        " "
                                        (mapconcat (lambda (tag) (format " --tag %s" tag)) tags " "))))
    (rspec-dired-verify-single)))
```

Same with the other commands.